src/Connection.php

```diff
@@ -1045,39 +1039,37 @@ public function transactional(Closure $func): mixed
     /**
      * Sets if nested transactions should use savepoints.
      *
+     * @deprecated No replacement planned
+     *
      * @throws Exception
      */
     public function setNestTransactionsWithSavepoints(bool $nestTransactionsWithSavepoints): void
```

Comment:
If we keep the method, we should throw if `false` is passed.

Reply:
Why deprecate it in the next major given that the dummy mode is already deprecated and is being removed?

Reply:
I see, so there are two breaking changes necessary:

1. Change the default mode from `false` to `true` in 4.0 (we're letting users opt in to the `true` mode in 3.x).
2. Remove the API for changing the mode in 5.0 (it will exist in 4.0 only for backward compatibility with 3.x).

Makes sense.